spec/unit/core/loggers/gd_logger_spec.rb

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/core/core'
+require 'securerandom'
+
+describe GoodData::GDLogger do
+  it "Has GoodData::GDLogger class" do
+    expect(GoodData::GDLogger).not_to be(nil)
+  end
+
+  let(:logger) { GoodData::GDLogger.new }
+
+  before(:each) do
+    logger.logging_on :logger, Logger.new(STDOUT)
+    logger.logging_on :errlog, Logger.new(STDERR)
+  end
+
+  describe "When logging is turned on" do
+    it "should output to stdout and stderr" do
+      expect { logger.info("Plain text") }.to output(/.*Plain text/).to_stdout_from_any_process
+      expect { logger.info("Plain text") }.to output(/.*Plain text/).to_stderr_from_any_process
+    end
+  end
+
+  describe "When using filter" do
+    it "it should write to stderr" do
+      expect { logger.add(Logger::INFO, "Plain text", nil, filters: Set[:logger]) }.to output.to_stderr_from_any_process
+    end
+
+    it "it should not write to stdout" do
```

Comment:
warning: Similar code found in gd_logger_spec.rb:28, gd_logger_spec.rb:32 (mass = 38) (flay)